libs/payment/api/src/lib/payment-provider/mollie-payment-provider.ts

```diff
@@ -0,0 +1,270 @@
+import {
+  BasePaymentProvider,
+  CheckIntentProps,
+  CreatePaymentIntentProps,
+  Intent,
+  IntentState,
+  InvoiceWithItems,
+  PaymentProviderProps,
+  WebhookForPaymentIntentProps,
+  WebhookResponse
+} from './payment-provider'
+import {logger} from '@wepublish/utils/api'
+import {PaymentState} from '@prisma/client'
+import createMollieClient, {
+  MollieClient,
+  SequenceType,
+  Payment,
+  PaymentMethod
+} from '@mollie/api-client'
+import MaybeArray from '@mollie/api-client/dist/types/types/MaybeArray'
+
+export interface MolliePaymentProviderProps extends PaymentProviderProps {
+  apiKey: string
+  webhookEndpointSecret: string
+  apiBaseUrl: string
+  methods?: string[]
+}
+
+const erroredPaymentIntent = {
+  intentID: 'error',
+  intentSecret: '',
+  intentData: 'error',
+  state: PaymentState.requiresUserAction,
+  errorCode: 'error'
+}
+
+type MolliePaymentMetadata = {
+  paymentID?: string
+  mail?: string
+}
+
+function mapMollieEventToPaymentStatus(event: string): PaymentState {
+  switch (event) {
+    case 'failed':
+    case 'expired':
+      return PaymentState.requiresUserAction
+    case 'open':
+    case 'authorized':
+    case 'pending':
+      return PaymentState.processing
+    case 'paid':
+      return PaymentState.paid
+    case 'canceled':
+      return PaymentState.canceled
+    default:
+      throw new Error(`Unhandled ${event} state received from mollie`)
+  }
+}
+
+function calculateAndFormatAmount(invoice: InvoiceWithItems) {
+  return `${(
+    invoice.items.reduce(
+      (prevItem, currentItem) => prevItem + currentItem.amount * currentItem.quantity,
+      0
+    ) / 100
+  ).toFixed(2)}`
+}
+
+export class MolliePaymentProvider extends BasePaymentProvider {
+  readonly webhookEndpointSecret: string
+  readonly mollieClient: MollieClient
+  readonly apiBaseUrl: string
+  readonly methods: MaybeArray<PaymentMethod> | undefined
+
+  constructor(props: MolliePaymentProviderProps) {
+    super(props)
+    this.webhookEndpointSecret = props.webhookEndpointSecret
+    this.apiBaseUrl = props.apiBaseUrl
+    this.mollieClient = createMollieClient({apiKey: props.apiKey})
+    this.methods = this.getPaymentMethode(props.methods)
+  }
+
+  getPaymentMethode(methods: string[] | undefined): MaybeArray<PaymentMethod> | undefined {
+    if (!methods) return undefined
```

Comment:
I suggest to always add curly brackets, even for single line blocks, so this does not happen: https://www.codecentric.de/wissens-hub/blog/curly-braces

There are also instances in L120 and L208